framework/src/kernels/NonlocalKernel.C

```diff
@@ -39,6 +39,58 @@ NonlocalKernel::NonlocalKernel(const InputParameters & parameters) :
 }
 
 void
+NonlocalKernel::computeJacobian()
+{
```

Comment:
Most of this code is copied directly out of Kernel. I wonder if there's a better way of implementing these algorithms in terms of the existing code. I realize that they are not 100% the same.